.github/CONTRIBUTING.rst

```diff
@@ -125,10 +125,99 @@ to work on. See also the development guidelines below.
 .. _dev-guidelines:
 
 Development guidelines
------------------------
+===========================
+
+It's good practice to create a separate development environment for your package development. Use your preferred
+system (or maybe integrated in your IDE) to setup a Python environment and see those docs to setup an environment
+(conda, pyenv, virtualenv,,...). Once created, you can install the package with all the developer dependencies
+using pip:
+
+::
+
+    pip install -e .[develop]
+
+Alternatively, if you are already familiar with ``tox``, run the ``dev`` tox command, which will create a ``venv`` with a
+development install of the package and it will register the environment as a ipykernel (for usage
+inside a jupyter notebook):
+
+::
+
+    tox -e dev
+
+For development purposes using conda, make sure to first run ``pip install -e .[develop]`` environment
+to prepare the development environment and install all development tools. (When using ``tox -e dev`` this
+is already done).
+
+When starting on the development of the ``pywaterinfo`` package, makes sure to be familiar with the following tools. Do
+not hesitate to ask the other contributors when having trouble using these tools.
+
+Pre-commit hooks
+----------------
+
+To ensure a more common code formatting and limit the git diff, make sure to install the pre-commit hooks. For example,
+`black <https://black.readthedocs.io/en/stable/index.html>`_. is used to make the code formatting as consistent as possible.The
+required dependencies are included in the development requirements installed when running ````pip install -e .[develop]``.
+
+.. warning::
+   Install and update the ``pre-commit`` hooks before your first git commit to the package!
+
+::
+
+    pre-commit install
+    pre-commit autoupdate
```

Comment:
Looks like this updates the versions to newer versions than .pre-commit-config.yaml. Is this what we want?

Reply:
As the autoupdate command actually updates the `pre-commit-config.yaml` file, I do think this is ok to do so. I just make sure the update is done for now (I should autoupdate myself ;-))

Reply:
Updated version in config done